File: 1318-minimum-flips-to-make-a-or-b-equal-to-c/1318-minimum-flips-to-make-a-or-b-equal-to-c.cpp
class Solution {
public:
    int minFlips(int a, int b, int c) {
        int ans = 0;
        for(int bit = 0; bit <= 30; bit++)
        {
            int mask = 1 << bit;
            if(c & mask)
            {
                if((a & mask) == 0 && (b & mask) == 0)
                    ans++;
            }
            else if((c & mask) == 0)
            {
                if(a & mask)
                    ans++;
                if(b & mask)
                    ans++;
            }
        }
        return ans;
    }
};

File: 1318-minimum-flips-to-make-a-or-b-equal-to-c/1318-minimum-flips-to-make-a-or-b-equal-to-c.go
func minFlips(a int, b int, c int) int {
    ans := 0
    for i := 0; i < 31; i++ {
        mask := 1 << i
        if c & mask > 0 {
            if (a & mask) | (b & mask) == 0 {
                ans++
            }
        } else if c & mask == 0 {
            if a & mask > 0 {
                ans++
            }
            if b & mask > 0 {
                ans++
            }
        }
    }
    return ans
}